tmt/steps/provision/artemis.py

```diff
@@ -268,6 +269,48 @@ class ProvisionArtemisData(ArtemisGuestData, tmt.steps.provision.ProvisionStepDa
     pass
 
 
+class ArtemisProvisionError(ProvisionError):
+    """
+    Artemis provisioning error.
+
+    For some provisioning errors, we can provide more context.
+    """
+
+    def __init__(
+            self,
+            message: str,
+            response: Optional[requests.Response] = None,
+            request_data: Optional[dict[str, Any]] = None,
+            *args: Any,
+            **kwargs: Any) -> None:
+        if response is not None:
+            message_components = [
+                message,
+                '',
+                f'Artemis API responded with code {response.status_code}: {response.reason}',
+                '',
+                f'Request: {response.request.url}',
+                ]
+
+            if request_data is not None:
+                message_components += [
+                    '',
+                    dict_to_yaml(request_data)
+                    ]
+
+            message_components += [
+                f'Response: {response.status_code} {response.reason}',
```

Comment:
The `{response.status_code} {response.reason}` part seems to be a duplicating line 290. Is it intentional to show the status code and reason twice?

Reply:
Not necessary, I dropped the duplicities.